### PR TITLE
SPM overview

### PR DESCRIPTION
**Pull request by alzix**

### CODE REVIEW

**Review by danny4478 — COMMENTED** [review]
Why does it include the porting guide (an old version of it)?

File: docs/api/security/psa.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Mbed PSA provides PSA API compliance for developing robust IoT applications and 
+allows to choose target type at later phase according to final application threat model.
+
+### Platform types
+Mbed PSA supports following platform types:
+- Non PSA platform - these are single core ARMv7-M targets. On these targets
+  Mbed PSA provides same PSA services exposing PSA APIs as it would on PSA targets. 
+  PSA emulation layer allows seamless software portability to more 
+  security oriented targets.
+- Asymmetric Multiprocessing (AMP) systems - Multi core ARMv7-M targets (e.g PSoC6
```
Comment (danny4478) [inline docs/api/security/psa.md]:
```suggestion
- Asymmetric Multiprocessing (AMP) systems - Multi core ARMv7-M targets (for example, PSoC6
```

Comment (danny4478) [reply docs/api/security/psa.md]: They don't use these kind of shortcuts in ARM documents

File: docs/api/security/psa.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+## Mbed PSA
```
Comment (danny4478) [inline docs/api/security/psa.md:1]: I think this should be an SPM overview, not a PSA overview

File: docs/api/security/psa.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+| SPM          | Secure Partition Manager            |
+| SPE          | Secure Processing Environment       |
+| NSPE         | Non-Secure Processing Environment   |
+| IPC          | Inter Process Communication         |
+| RoT          | Root Of Trust                       |
+
+
+### Overview
+Mbed PSA provides essential root of trust services and infrastructure for developing robust IoT applications.
+When mbed-os is running on PSA Security Model compliant target, Mbed PSA helps to protect cryptographic assets, credentials, and critical code sections by providing an isolation between a Secure Processing Environment (SPE) and a Non-Secure Processing Environment (NSPE). The isolation is managed by Secure Partition Manager (SPM) which utilizes unique HW features available on the target. SPM provides standardized IPC APIs which abstract the fact that partitions could be living inside a virtualized environment (v8M, TEE on Cortex-A), or another chip.
```
Comment (danny4478) [inline docs/api/security/psa.md:17]: I think we should use Mbed OS (not mbed os)

File: docs/api/security/psa.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+Mbed PSA bridges the differences between PSA and Non-PSA targets for application developers, allowing to use same standard PSA APIs on both target types.
```
Comment (danny4478) [inline docs/api/security/psa.md:19]: What does this sentence mean? It is not clear to me.

File: docs/api/security/psa.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+## Mbed PSA
+
+### Terms and Abbreviations
+
+| Abbreviation | Description                         |
```
Comment (danny4478) [inline docs/api/security/psa.md]:
```suggestion
| Term         | Meaning                             |
```

File: docs/api/security/psa.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+| IPC          | Inter Process Communication         |
+| RoT          | Root Of Trust                       |
+
+
+### Overview
+Mbed PSA provides essential root of trust services and infrastructure for developing robust IoT applications.
+When mbed-os is running on PSA Security Model compliant target, Mbed PSA helps to protect cryptographic assets, credentials, and critical code sections by providing an isolation between a Secure Processing Environment (SPE) and a Non-Secure Processing Environment (NSPE). The isolation is managed by Secure Partition Manager (SPM) which utilizes unique HW features available on the target. SPM provides standardized IPC APIs which abstract the fact that partitions could be living inside a virtualized environment (v8M, TEE on Cortex-A), or another chip.
+
+Mbed PSA bridges the differences between PSA and Non-PSA targets for application developers, allowing to use same standard PSA APIs on both target types.
+Mbed PSA provides PSA API compliance for developing robust IoT applications and
```
Comment (danny4478) [inline docs/api/security/psa.md:20]: There is a new line here for some reason...

File: docs/api/security/psa.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+### Overview
+Mbed PSA provides essential root of trust services and infrastructure for developing robust IoT applications.
+When mbed-os is running on PSA Security Model compliant target, Mbed PSA helps to protect cryptographic assets, credentials, and critical code sections by providing an isolation between a Secure Processing Environment (SPE) and a Non-Secure Processing Environment (NSPE). The isolation is managed by Secure Partition Manager (SPM) which utilizes unique HW features available on the target. SPM provides standardized IPC APIs which abstract the fact that partitions could be living inside a virtualized environment (v8M, TEE on Cortex-A), or another chip.
+
+Mbed PSA bridges the differences between PSA and Non-PSA targets for application developers, allowing to use same standard PSA APIs on both target types.
+Mbed PSA provides PSA API compliance for developing robust IoT applications and 
+allows to choose target type at later phase according to final application threat model.
+
+### Platform types
+Mbed PSA supports following platform types:
```
Comment (danny4478) [inline docs/api/security/psa.md]:
```suggestion
Mbed PSA supports the following platform types:
```

File: docs/api/security/psa.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+- Non PSA platform - these are single core ARMv7-M targets. On these targets
```
Comment (danny4478) [inline docs/api/security/psa.md:25]: Why is this strange lines breaking?

File: docs/api/security/psa.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  Mbed PSA provides same PSA services exposing PSA APIs as it would on PSA targets. 
+  PSA emulation layer allows seamless software portability to more 
+  security oriented targets.
+- Asymmetric Multiprocessing (AMP) systems - Multi core ARMv7-M targets (e.g PSoC6
+  featuring CM4 and CM0+ cores). On these targets one of the cores is dedicated to PSA usage only and implements SPE. Mbed PSA provides PSA APIs proxy implementation on non-secure core, which redirect execution to SPE.
```
Comment (danny4478) [inline docs/api/security/psa.md]: "On these targets one of the cores is dedicated to PSA usage only and implements SPE" - This is not correct.
It is SPM that is running on one target only. PSA, which is not a module but more of a model consists of several specifications, is not running anywhere :)

Comment (danny4478) [inline docs/api/security/psa.md]:
```suggestion
  featuring CM4 and CM0+ cores). On these targets one of the cores is dedicated to PSA usage only and implements SPE. Mbed PSA provides PSA APIs proxy implementation on non-secure core, which redirects execution to SPE.
```

**Review by danny4478 — CHANGES_REQUESTED** [review]

File: docs/api/security/psa.md (hunk 2 above)
Comment (alzix) [reply docs/api/security/psa.md:1]: SPM is a part of it. Perhaps we can add a section elaborating about SPM

File: docs/api/security/psa.md (hunk 4 above)
Comment (alzix) [reply docs/api/security/psa.md:19]: Top level PSA APIs will be available on both platforms. So you can develop your SW and then switch to a different PSA target according to your threat model

File: docs/api/security/psa.md (hunk 3 above)
Comment (alzix) [reply docs/api/security/psa.md:17]: done

File: docs/api/security/psa.md (hunk 6 above)
Comment (alzix) [reply docs/api/security/psa.md:20]: it was for the sake of keeping lines short.
i went the  whole way this time. checkout updated version.

File: docs/api/security/psa.md (hunk 8 above)
Comment (alzix) [reply docs/api/security/psa.md:25]: 80 column

**Review by danny4478 — COMMENTED** [review]
I am a little bit confused.
We still have to add a Reference API document which also comes under api directory.
We need to agree on locations of all of these documents.

**Pull request closed without merging**